### PR TITLE
Fixes #16744 - sync plan response format change

### PR DESCRIPTION
**Pull request by amitkarsale**
sync plan create api returns 500 for missing or improperly formatted sync date, It should return a bad request.

### CODE REVIEW

**Comment by jlsherrill** [comment]
Hi @amitkarsale,  we try to use 1 commit per PR and fix 1 issue in each PR.  

Can you squash your commits on 16744 together?  (and it looks like you've accidentally got a 3rd commit from a different PR in this pr)

**Comment by jlsherrill** [comment]
@amitkarsale   if you set your Katello organization visibility public tests will run automatically for you :)
https://help.github.com/articles/publicizing-or-hiding-organization-membership/

**Comment by amitkarsale** [comment]
[test]

**Comment by amitkarsale** [comment]
@jlsherrill merged to single commit, can we verify it now.

File: app/views/katello/api/v2/content_views/base.json.rabl [hunk 1]
```diff
@@ -5,9 +5,12 @@ attributes :composite
 attributes :repository_ids
 attributes :component_ids
 attributes :default
-attributes :next_version
 attributes :force_puppet_environment
 
+node :next_version do |content_view|
+  content_view.next_version.to_f.to_s
+end
+
```
Comment (jlsherrill) [inline app/views/katello/api/v2/content_views/base.json.rabl]: I don't think you intended this change to be in this PR, did you?

File: app/controllers/katello/api/v2/sync_plans_controller.rb [hunk 2]
```diff
@@ -47,10 +47,6 @@ def show
     def create
       sync_date = sync_plan_params[:sync_date].to_time(:utc)
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/sync_plans_controller.rb]: it looks like if 'sync_date' isn't passed in at all (and so sync_plan_params[:sync_date]) is nill, the controller still returns a 500 ISE. 

You might change that to:

`sync_date = sync_plan_params[:sync_date].try(:to_time, :utc)`

File: app/models/katello/sync_plan.rb [hunk 3]
```diff
@@ -31,7 +31,7 @@ class SyncPlan < Katello::Model
     scoped_search :on => :enabled, :complete_value => true
 
     def validate_sync_date
-      errors.add :base, _("Start Date and Time can't be blank") if self.sync_date.nil?
+      fail HttpErrors::BadRequest, _("Sync Date format is incorrect.") if self.sync_date.nil?
     end
```
Comment (jlsherrill) [inline app/models/katello/sync_plan.rb]: You might just get rid of this method altogether and replace it with:
```
    validates :sync_date, :presence => true
```

towards the top of the model.  I'm not sure why this even exists :)

Comment (jlsherrill) [inline app/models/katello/sync_plan.rb]: exists as a method i mean.

File: app/views/katello/api/v2/content_views/base.json.rabl (hunk 1 above)
Comment (amitkarsale) [inline app/views/katello/api/v2/content_views/base.json.rabl]: opps, I goofed the file changes from other branch to this one.

**Comment by ehelms** [comment]
[test]

**Comment by jlsherrill** [comment]
@amitkarsale any update on this?

**Comment by amitkarsale** [comment]
looking in this

**Comment by beav** [comment]
@amitkarsale it looks like this needs a rebase

**Comment by beav** [comment]
Is this PR still active? It has not been updated in a year.

**Comment by johnpmitsch** [comment]
Its been a while with no activity or response on this PR, so I'm going to close, but feel free to reopen if the PR is still valid :)

**Pull request closed without merging**